### ISSUE TITLE
How to handle idle timeout in RemoteP2P Module

### ISSUE THREAD

**Issue by sreeraaman** [ls1intum/jReto#2]
We are experimenting with the RemoteP2P module which essentially uses WebSockets. The issue here is that the RemoteP2P Module is deployed as war in a OSGi Container (Karaf + Jetty 9). Jetty by default has a connection timeout of 5 minutes after which it closes the websocket connection. In the RemoteP2PModule class, the onClose() gets invoked. This essentially stops the p2p infrastructure from functioning then onwards.

The current implementation of the onClose() method in the RemoteP2PModule is as follows:

```
    this.webSocket = null;
    this.startedDiscoverySocket = false;
    this.advertiserHandler.onAdvertisingStopped(this);
    this.browserHandler.onBrowsingStopped(this, null);  
```

However, to ensure that we handle idle timeouts properly, we had fixed this code as follows:

```
            this.webSocket = null;
    this.startedDiscoverySocket = false;
    this.advertiserHandler.onAdvertisingStopped(this);
    this.browserHandler.onBrowsingStopped(this, null);  
    if (closeReason.getCloseCode().equals(CloseReason.CloseCodes.GOING_AWAY)){
        System.out.println("Reconnecting  ...");
        startDiscoveryWebSocket();
    }
```

Just wanted to understand if this would impact else where or is it okay to have this. I can create a pull request if you are convinced about the fix.

best regards
Sriraman.

**Comment by krusche** [comment]
Sounds reasonable! Can you open a pull request with these changes?
I would be happy to accept it

**Comment by sreeraaman** [comment]
Thanks for the quick response. I have a couple of other changes to the RemoteP2PServer only in the pom.xml. The changes are mainly to do with the dependencies for getting it deployed in a non-java ee environment such as apache karaf. Also, I have the maven bundle plugin configuration which would make jreto a proper osgi bundle. Will create the pull request shortly.

**Comment by krusche** [comment]
Was addressed by Pull Request #3
